### ISSUE TITLE
How to join cluster

### ISSUE THREAD

**Issue by seedotlee** [Mashape/kong#1592]
### Summary

I have 2 servers ,call them ser1 and ser2.

I started ser1 with config

```
pg_host = 1.1.1.1 #ser1's IP

cluster_listen = 1.1.1.1:7946  # ser1's IP
cluster_listen_rpc = 127.0.0.1:7373 
cluster_profile = lan
```

After, I started ser2 with config

```
pg_host = 1.1.1.1 #ser1's IP

cluster_listen = 2.2.2.2:7946  # ser2's IP
cluster_listen_rpc = 127.0.0.1:7373 
cluster_profile = lan
```

Two servers can visited, but I get `http://1.1.1.1:8001/cluster` just 1 cluster.

Using cluster should I pay attention to ？

### Additional Details & Logs

- Kong version (`0.9.0`)
- Operating System ubuntu 14.04

**Comment by seedotlee** [comment]
I used `serf` to join 2 servers all going well

**Comment by seedotlee** [comment]
I'm solving the problem.

Becuase ser2 is ser1's copy , so the serf.id is the same with ser1.

The serf.id is not change with kong's start that not good enough.

**Comment by thefosk** [comment]
@seedotlee what do you mean by "serf.id" ?

**Comment by seedotlee** [comment]
`/usr/local/kong/serf/serf.id`

**Comment by thefosk** [comment]
@seedotlee you would need to specify different `prefix` for each Kong instance. Ie:

```
prefix=/tmp/kong1
pg_host = 1.1.1.1 #ser1's IP

cluster_listen = 1.1.1.1:7946  # ser1's IP
cluster_listen_rpc = 127.0.0.1:7373 
cluster_profile = lan
```

and

```
prefix=/tmp/kong2
pg_host = 1.1.1.1 #ser1's IP

cluster_listen = 1.1.1.1:7946  # ser1's IP
cluster_listen_rpc = 127.0.0.1:7373 
cluster_profile = lan
```

**Comment by seedotlee** [comment]
@thefosk 
Uh, I started in 2 machines, there is need different `prefix`?

**Comment by thibaultcha** [comment]
It could be that you have the same hostname for both machines, and that the random part of this line: https://github.com/Mashape/kong/blob/master/kong/cmd/utils/prefix_handler.lua#L218 is actually not so random since from what I can tell, the random number generator is not seeded when invoking it in the CLI, hence both nodes generate the same uuid. If you could add some debug statement right there and confirm with us that they are both generating the same uuid (and that you have the same hostname on both nodes), then I would qualify this as a bug. The fix would be to - properly - seed the random number generation. This is definitely tricky with Lua but once done right, the random numbers are of good quality (with the LuaJIT implementation, especially).

**Comment by thefosk** [comment]
@thibaultcha this could be a problem. Do we need a specific patch for this in 0.9.2?

We should also check everywhere where `utils.random_string()` is been used, because every caller expects the returning value to be a universally unique string.

**Comment by thibaultcha** [comment]
Should be fixed with #1641, to be released in 0.9.2, thanks for reporting.
